### PR TITLE
fix(CardContent): allows users to add columnSpan as prop

### PR DESCRIPTION
**Pull request by slhay28**
## Description

Adds a `metadataColumnSpan` and `tileColumnSpan` prop so users can control the size of cardContent

## Changes

- `metadataColumnSpan` prop
- `tileColumnSpan` prop 

## Testing

- change values in these added properties to see component change

### CODE REVIEW

File: src/packages/solid/components/CardContent/CardContent.tsx [hunk 1]
```diff
@@ -143,42 +152,29 @@ const CardContent: Component<CardContentProps> = props => {
         states={{ collapsed: collapsedTile() }}
         style={[styles.TileContainer.tones[props.tone ?? styles.tone], styles.TileContainer.base]}
         width={
-          styles.TileContainer.tones[props.tone ?? styles.tone]?.width ?? styles.TileContainer.base.width
-        }
-        height={
-          styles.TileContainer.tones[props.tone ?? styles.tone]?.height ?? styles.TileContainer.base.height
+          props.tileColumnSpan ? getWidthByColumnSpan(props.tileColumnSpan) : styles.TileContainer.base.width
         }
+        height={props.height || styles.TileContainer.base.height}
```
Comment (anthony9187) [inline src/packages/solid/components/CardContent/CardContent.tsx:157]: use `??` here, height could potentially be zero(yes it's a weird edge case)

File: src/packages/solid/components/CardContent/CardContent.tsx [hunk 2]
```diff
@@ -143,42 +152,29 @@ const CardContent: Component<CardContentProps> = props => {
...
       {/* Right side of component- where metadata is  */}
       <Show when={!collapsedTile()}>
         <View
           style={[styles.MetadataContainer.tones[props.tone ?? styles.tone], styles.MetadataContainer.base]}
+          width={
+            props.metadataColumnSpan
+              ? getWidthByColumnSpan(props.metadataColumnSpan)
+              : styles.MetadataContainer.base.width
+          }
+          height={props.height || styles.MetadataContainer.base.height}
```
Comment (anthony9187) [inline src/packages/solid/components/CardContent/CardContent.tsx:170]: same here

File: src/packages/solid/utils/getWidthByColumnSpan.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import theme from 'theme';
+import { getWidthByUpCount } from './getWidthByUpCount.js';
+export function getWidthByColumnSpan(columnSpan: number) {
+  const columnCount = theme.layout.columnCount;
+  const gutterX = theme.layout.gutterX;
+
+  return (getWidthByUpCount(columnCount) ?? 1) * columnSpan + gutterX * (columnSpan - 1);
```
Comment (anthony9187) [inline src/packages/solid/utils/getWidthByColumnSpan.ts:24]: do we need the `??` here? getWidthByUpCount should only ever return numeric values

Comment (slhay28) [reply src/packages/solid/utils/getWidthByColumnSpan.ts:24]: I agree, but I get a possibly undefined ts error so I wasnt sure

**Review by anthony9187 — APPROVED** [review]

**Pull request merged**